addons/product/models/product_product.py

```diff
@@ -801,5 +801,6 @@ def _get_contextual_discount(self):
             fields.Datetime.now(),
         )
```

Comment:
The fix seems ok, we're indeed comparing rounded and not rounded prices here, which is a mistake.
But is it necessary to round the discounted price?
Did you try disabling the rounding with `round=False` in the `_convert` call above to compare not rounded prices instead of comparing rounded ones ?

Reply:
Thanks for the feedback. It was actually an oversight on my part to name that variable "discounted_price", it's just the result of _get_contextual_price before any discounts are applied.
Adding round=False does indeed fix the issue as well. Since it seems like a more elegant solution, I will update the commits to use that instead.